src/app/calendar/calendar.component.ts

```diff
@@ -1,106 +1,180 @@
+        private toastr: ToastrService,
+        private reservationService: ReservationService
+    ) {
+
+        this.updateViewBasedOnWidth();
+    }
+
+    mapEvents(reservations: Reservation[]): void {
+        this.events = reservations.map(reservation => new CalendarEvent(
+            [
+                {
+                    id: reservation.id,
+                    title: this.selectedUser!.firstName,
+                    start: reservation.startDateTime,
+                    end: reservation.endDateTime
+                }
+            ],
+            'blue',
+            'white'
+        ));
+    }
+
+    ngOnInit() {
+        this.personSelectForm = new FormGroup({
+            'favourite-user': new FormControl(null, Validators.required)
+        });
+
+        this.getFavoriteUsers();
+        this.updateViewBasedOnWidth();
+    }
+
+    getFavoriteUsers(): void {
+        this.favoriteUserService.getFavoriteColleagues().subscribe(
+            (response) => {
+                this.favoriteColleagues = response.payload;
+            },
+            (error) => {
+                this.toastr.error("Probeer het later nog een keer", "Fout bij ophalen van favoriete collega's")
+            }
+        );
+    }
+
+    updateViewBasedOnWidth() {
+        let initialView!: string;
+        if (window.innerWidth < 768) {
+            initialView = 'timeGridDay';
+        } else if (window.innerWidth >= 768 && window.innerWidth < 992) {
+            initialView = 'timeGridWeek';
+        } else {
+            initialView = 'dayGridMonth';
+        }
+
+        this.calendarOptions.set({
+            ...this.calendarOptions(),
+            initialView: initialView
+        });
+    }
+
+
+    handleEventClick(clickInfo: EventClickArg) {
+        if (confirm(`Are you sure you want to delete the event '${clickInfo.event.title}'`)) {
+            clickInfo.event.remove();
+        }
+    }
 
-  handleEventClick(clickInfo: EventClickArg) {
-    if (confirm(`Are you sure you want to delete the event '${clickInfo.event.title}'`)) {
-      clickInfo.event.remove();
+    handleEvents(events: EventApi[]) {
+        this.currentEvents.set(events);
+        this.changeDetector.detectChanges();
     }
-  }
 
-  handleEvents(events: EventApi[]) {
-    this.currentEvents.set(events);
-    this.changeDetector.detectChanges();
-  }
+    onChangePerson() {
+        this.selectedUser = this.personSelectForm.get('favourite-user')?.value;
+        let userId: string = this.selectedUser!.id!;
+
+        this.reservationService.getReservationsByUserId(userId).subscribe(
+            (reservations) => {
+                this.mapEvents(reservations);
+
+                this.calendarOptions.set({
+                    ...this.calendarOptions(),
+                    eventSources: this.events
+                });
+            },
+            (error) => {
+                console.error('Error fetching reservations: ', error);
```

Comment:
kun je hiervan een popup maken ipv een console error

Reply:
Dit is niet van mij